stan/math/prim/meta/compiler_attributes.hpp

```diff
@@ -4,11 +4,39 @@
 #ifdef __GNUC__
 #define likely(x) __builtin_expect(!!(x), 1)
 #define unlikely(x) __builtin_expect(!!(x), 0)
+#ifdef __has_attribute
+#if __has_attribute(noinline) && __has_attribute(cold)
+/**
+ * Functions tagged with this attribute are not inlined and moved
+ *  to a cold branch to tell the CPU to not attempt to pre-fetch
+ *  the associated function.
+ */
 #define STAN_COLD_PATH __attribute__((noinline, cold))
 #else
+#define STAN_COLD_PATH
+#endif
+#else
+#define STAN_COLD_PATH
+#endif
+#else
```

Comment:
These changes seem unrelated to what this PR is doing.

Reply:
This is a little bit of a fix, if someone is using either not gcc or clang then we are not guaranteed `__has_attribute` is defined. So to compile on those we want to make sure that macro is defined so that we can then check whether the compiler has the attributes `noinline` and `cold`

Reply:
Ok, but you could simplify this into:
```suggestion
#ifdef __has_attribute
#if __has_attribute(noinline) && __has_attribute(cold)
/**
 * Functions tagged with this attribute are not inlined and moved
 *  to a cold branch to tell the CPU to not attempt to pre-fetch
 *  the associated function.
 */
#define STAN_COLD_PATH __attribute__((noinline, cold))
#endif
#ifndef STAN_COLD_PATH
#define STAN_COLD_PATH
#endif
```